### ISSUE TITLE
CA2213 warning for not disposing a timer, but I do.

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn-analyzers#3820]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/1100166/i-get-a-compiler-warning-for-not-disposing-a-timer.html)._

---
I get the following warning from the compiler:

Warning	CA2213	'SplashScreen' contains field 'splashScreenTimer' that is of IDisposable type 'Timer', but it is never disposed. Change the Dispose method on 'SplashScreen' to call Close or Dispose on this field.	ChumsMailer	C:\gitrepos\ChumsMailer\Dialogs\SplashScreen.cs

I do have a call to Dispose in my code for the timer which my log and debugger confirm is being called. I attach a screen shot of the relevant code.

---
#### Visual Studio Feedback System on 7/3/2020, 01:42 AM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

#### Visual Studio Feedback System on 7/6/2020, 10:19 AM: 

<p>Thank you for sharing your feedback! Our teams prioritize action on product issues with broad customer impact. See details at: <a target='_blank' href="https://docs.microsoft.com/en-us/visualstudio/ide/report-a-problem?view=vs-2019#faq">https://docs.microsoft.com/en-us/visualstudio/ide/report-a-problem?view=vs-2019#faq</a>. In case you need answers to common questions or need assisted support, be sure to use <a target='_blank' href="https://visualstudio.microsoft.com/vs/support/">https://visualstudio.microsoft.com/vs/support/</a>. We’ll keep you posted on any updates to this feedback.</p>

---
### Original Solutions
(no solutions)

**Comment by Evangelink** [comment]
@mavasani The user is talking about some screenshot but I don't see any. If you do see some on your side, would you mind sharing it here? 

With only the description there is no way to actually see if there is any problem going-on.

**Comment by martincostello** [comment]
I've updated a number of repos from .NET 5 preview 7 to preview 8, and now am getting a number of false positives for the CA2213 rule:

  * https://github.com/martincostello/sqllocaldb/pull/234/commits/8ef6ce97240f2041033c9b2c91793ba120bcb670
  * https://github.com/martincostello/alexa-london-travel-site/pull/599/commits/27d5aec6c3aaae048892e593a16dc8440f42d293

**Comment by Evangelink** [comment]
@martincostello That's interesting... Looking at the files this is clearly a FP but I am having some trouble to understand what's going wrong. Could you confirm that the following snippet doesn't raise the issue:

```csharp
public sealed class DisposableClass : IDisposable
    {
        public void Dispose() { }
    }

    public sealed class Class1 : IDisposable
    {
        private readonly DisposableClass _d = new DisposableClass();
        private bool _disposed;

        public void Dispose()
        {
            Dispose(true);
            GC.SuppressFinalize(this);
        }

        private void Dispose(bool disposing)
        {
            if (!_disposed)
            {
                if (disposing)
                {
                    // Dispose of managed resources
                    _d.Dispose();
                }

                _disposed = true;
            }
        }
    }
```

**Comment by martincostello** [comment]
I forgot to come back to this issue and update it, but these false positives resolved themselves after updating to 5.0 RC1.

**Comment by Evangelink** [comment]
I will close this issue then as there is no feedback on the original issue.
